force-app/main/default/classes/Homework.cls

```diff
@@ -18,31 +18,63 @@ public with sharing class Homework
     // 1. Take a List of integers and return the largest value using a loop
     public Integer maxInListLoop(List<Integer> values)
     {
+        Integer max = values[0];
 
+        for (Integer j = 1; j < values.size() - 1; j++) {
+
+            max = (values[j] > max) ? values[j] : max;
+        }
+        
+        return max;
     }
 
     // 2. Take a List of integers and return the largest value without using a loop.
     public Integer maxInList(List<Integer> values)
     {
-
+        Integer last = values.size() - 1;
+        values.sort();
+        System.debug(values);
+        Integer max = values[last];
+        System.debug('max => ' + max);
+        return max;
     }
 
     // 3. Take a list of Integers and reverse the order
     public List<Integer> reverse(List<Integer> values)
     {
+        List<Integer> reverseValues = new List<Integer>();
 
+        for (Integer i = values.size() - 1; i >= 0; i--) {
+            reverseValues.add(values[i]);
+        }
+        return reverseValues;
     }
 
     // 4. Take a list of Strings and extract every third element into a new list that is returned. 
     //    Extra credit for handling that the list passed in may not be exactly divisible by 3.
     public List<String> everyThird(List<String> originals)
     {
+        List<String> everyThird = new List<String>();
+        Integer divisibleLength = originals.size() - Math.mod(originals.size(), 3);
+        System.debug(LoggingLevel.ERROR, '\n####  divisibleLength => ' + divisibleLength); //TODO: Remove this debug statement
+
+        for (Integer i = 0; i < divisibleLength; i += 3) 
+        {
+            everyThird.add(originals[i]);
+            System.debug('Adding originals[' + i + '] => ' + originals[i]);
+        }
+        
```

Comment:
This approach is fine, but I don't think you need to worry about the divisible length. If the collection is 10 elements, you will change `i` from 9 to 12, so just checking if `i` is greater than the collection length will work.